### DISCUSSION TITLE
What does `model_validate()`'s `from_attributes` parameter actually do?

### DISCUSSION THREAD

**Discussion by mjog** [pydantic/pydantic#8933] [Question]
Looking at the [API docs](https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel.model_validate) for `model_validate`, it says that function "Validate a pydantic model instance", which isn't terribly helpful, but I understand that it roughly "gets model field values from the attributes of the given object".

However what then does `from_attributes` actually do? By default it is `None`, so I assume that evaluates to `False`, but it still seems to get the values from objects regardless. Indeed, setting it true or false doesn't seem to actually change anything, and in any case if the function does not get the model's field values from the attributes of the given object when it is set to false, where else does it get them from?

**Comment by Antsthebul** [comment]
Came across this while searching and figured I'd answer. This seems to allow models with partially matching fields to be constructed into a new model. Basically what I had was
```
class ModelA(BaseModel):
    name: str

class ModelB(BaseModel):
    name: str
    age: int

>> model_b = ModelB(name="foo", age=20)
>> model_a = ModelA.model_validate(model_b) # fails w/ Input must be valid dictionary or instance of ModelA
>> model_a = ModelA.model_validate(model_b, from_attributes=False) # <- Good
```

Same logic is discussed a litter more in detail here https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict.from_attributes .

**Reply by hobofan** [reply]
The last line in the example code is wrong. It actually works the inverse, and for it to accept a different instance type, you have to set `from_attributes=True`:
```
model_a = ModelA.model_validate(model_b, from_attributes=True)
```
